tf2onnx/optimizer/identity_optimizer.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""Identity Optimizer.
+   Remove useless Identity node in graphs including subgraphs, but does not hurt model output names.
+"""
+
+from __future__ import unicode_literals
+import logging
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("tf2onnx.optimizer.identity_optimizer")
+
+
+# pylint: disable=logging-not-lazy,unused-argument,missing-docstring
+# FIXME:
+# pylint: disable=unused-variable
+
+
+class IdentityOptimizer(object):
+    """Identity Optimizer."""
+
+    def __init__(self, main_graph, output_names, debug=False):
+        self._g = main_graph
+        self._output_names = [name.split(":")[0] for name in output_names]
+        self._debug = debug
+
+    def optimize(self):
+        previous_counter = self._g.dump_node_statistics()
+        self._optimize_recursively(self._g)
+        current_counter = self._g.dump_node_statistics()
+        identity_cnt = current_counter["Identity"]
+        current_counter.subtract(previous_counter)
+        log.info(" %d identity op(s) left, ops diff after identity optimization: %s", identity_cnt, current_counter)
+
+    def _optimize_recursively(self, g):

```

Comment:
seems to be a common requirement to traverse graph recursively, consider something like ```g.traverse_recursively(lamda g: func)```?

Reply:
optional

Reply:
I would prefer to do this later. there are a few places to replace, which require some careful validation.